File: L1/java/Main.java
public class Main
{
    private Main() throws InstantiationError
    {
        throw new InstantiationError("This is a static class!");
    }

    public static void main(final String[] args)
    {
        if (args.length < 2)
        {
            System.out.println("Program requires at least 2 args to work properlym, got " + args.length);
            return;
        }

        try
        {
            final int endOfRange = Integer.parseInt(args[0]);
            final PrimesGenerator primesGenerator = new PrimesGenerator(endOfRange);

            for (int i = 1; i < args.length; ++i)
            {
                try
                {
                    final int primeIndex = Integer.parseInt(args[i]);
                    System.out.println(primeIndex + " --> " + primesGenerator.getPrime(primeIndex));
                }
                catch(final IllegalArgumentException e)
                {
                    System.out.println(args[i] + " --> " + e.toString());
                }
            }
        }
        catch(final IllegalArgumentException e) // NumberFormatException is a subclass of alternative IllegalArgumentException
        {
            System.out.println(e.toString());
        }
    }
}

File: L1/cpp/app/main.cpp
#include <primes.hpp>
#include <valueConverter.hpp>
#include <primesGenerator.hpp>

#include <iostream>

int main(int argc, const char* const argv[])
{
    if (argc < 3)
    {
        std::cerr << "Program requires at least 2 args to work properlym, got " << (argc - 1) << std::endl;
        return 0;
    }

    try
    {
        const int endOfRange = ValueConverter::stringToVal<int>(std::string(argv[1]));
        const PrimesGenerator primesGenerator(endOfRange);

        for (size_t i = 2; i < static_cast<size_t>(argc); ++i)
        {
            try
            {
                const size_t primeIndex = ValueConverter::stringToVal<size_t>(std::string(argv[i]));
                const int prime = primesGenerator[primeIndex];

                std::cout << std::to_string(primeIndex) << " --> " << std::to_string(prime) << std::endl;
            }
            catch (const std::invalid_argument &e)
            {
                std::cerr << "Error: " << e.what() << std::endl;
            }
        }
    }
    catch (const std::invalid_argument& e)
    {
        std::cerr << "Error: " << e.what() << std::endl;
    }

    return 0;
}
